subsys/bluetooth/controller/ll_sw/ull_llcp_common.c

```diff
@@ -471,8 +472,21 @@ static void lp_comm_complete(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t
 		break;
 #endif /* CONFIG_BT_CTLR_MIN_USED_CHAN && CONFIG_BT_PERIPHERAL */
 	case PROC_VERSION_EXCHANGE:
-		lp_comm_ntf_complete_proxy(conn, ctx,
-			(ctx->response_opcode == PDU_DATA_LLCTRL_TYPE_VERSION_IND));
+		if (ctx->response_opcode == PDU_DATA_LLCTRL_TYPE_VERSION_IND) {
+			if (ctx->node_ref.rx || llcp_ntf_alloc_is_available()) {
+				/* Either this is a piggy-back or there is a NTF node avail */
+				lp_comm_ntf(conn, ctx);
+				llcp_lr_complete(conn);
+				ctx->state = LP_COMMON_STATE_IDLE;
+			} else {
+				/* Handle procedure TO JUST in case we end up waiting 'forever' */
+				llcp_lr_prt_restart(conn);
```

Comment:
@thoh-ot you refer to this one, right? 
It is un purpose in that I use the TO as way to ensure that we will complete the procedure on TO in case we wait forever for a NTF buffer
It is in a way a hack to ensure we don't get stuck here on a locally initiated VERSION_IND that has NOT tx - ie where a VERSION IND has already been sent

Reply:
Partly, I was talking about when you have reset the TO and then go to the LP_COMMON_STATE_WAIT_NTF_AVAIL state. In that state there is a guard condition to only continue if a NTF buffer is available. In an (very unlikely) case that could take "forever" also, but the TO isn't reset in this state handler - is that on purpose?